source/Planet.cpp

```diff
@@ -205,10 +205,12 @@ void Planet::Load(const DataNode &node, Set<Wormhole> &wormholes)
 							defenseFleets.clear();
 							resetFleets = false;
 						}
-						defenseFleets.insert(defenseFleets.end(),
-								grand.Size() >= 3 ? grand.Value(2) : 1,
-								GameData::Fleets().Get(grand.Token(1))
-						);
+						const Fleet *fleet = GameData::Fleets().Get(grand.Token(1));
+						if(fleet->IsValid())
+							defenseFleets.insert(defenseFleets.end(),
+									grand.Size() >= 3 ? grand.Value(2) : 1,
+									fleet
+							);
```

Comment:
we should prob add an else saying "Warning: skipped non valid tribute fleet" or smth?

Reply:
Testing the warning and... forgot that data is loaded alphabetically so that any fleet in a file after "map planets" is treated as undefined. We need to load the fleets anyway and check if they're valid when demanding tribute.

Reply:
or you just add a finishloading method like part of the game has

Reply:
Yeah, also thought of this, and FinishLoading already exists for Planet class. Moving fleet validation there works, but the warning is displayed 17 times for a single planet and idk why (see the last commit).

Reply:
each time there is an event change this happens I think ~~but having 17 warnings isnt such a big deal~~